From ab254ddf9d4f7b87efc4513813fd7fccb4501901 Mon Sep 17 00:00:00 2001
From: tstegema <tstegema@90b65887-3827-0410-9a23-83215b262276>
Date: Thu, 29 Jan 2015 20:55:38 +0000
Subject: [PATCH 0643/1077] MANTIS 0034055 platform.idl V3.0 added struct
 PlatformEnvironment for platform settings added "PlatformEnvironment
 getEnvironment()" to PlatformIF to receive platform settings added
 airlineCode and applicationName to struct mApplicationProperties

CUSS.PAYMENT.XSD
set elements name, pnr, date, flight-number, departure-city, arrival-city, card-pan to xs:string


git-svn-id: svn://localhost/SelfServiceCommon/trunk@816 90b65887-3827-0410-9a23-83215b262276
---
 .../Massai/idl/Cuss/CUSS.PAYMENT.XSD             | 14 +++++++-------
 SelfServiceCommon/Massai/idl/Massai/platform.idl | 16 +++++++++++++---
 2 files changed, 20 insertions(+), 10 deletions(-)

diff --git a/SelfServiceCommon/Massai/idl/Cuss/CUSS.PAYMENT.XSD b/SelfServiceCommon/Massai/idl/Cuss/CUSS.PAYMENT.XSD
index c79b6ccc..9f633c20 100644
--- a/SelfServiceCommon/Massai/idl/Cuss/CUSS.PAYMENT.XSD
+++ b/SelfServiceCommon/Massai/idl/Cuss/CUSS.PAYMENT.XSD
@@ -307,32 +307,32 @@
                                             </xs:annotation>
                                             <xs:complexType>
                                                 <xs:sequence>
-                                                    <xs:element name="name" minOccurs="0">
+                                                    <xs:element name="name" type="xs:string" minOccurs="0">
                                                         <xs:annotation>
                                                             <xs:documentation>Passenger name</xs:documentation>
                                                         </xs:annotation>
                                                     </xs:element>
-                                                    <xs:element name="pnr" minOccurs="0">
+                                                    <xs:element name="pnr" type="xs:string" minOccurs="0">
                                                         <xs:annotation>
                                                             <xs:documentation>Passenger name record</xs:documentation>
                                                         </xs:annotation>
                                                     </xs:element>
-                                                    <xs:element name="date" minOccurs="0">
+                                                    <xs:element name="date" type="xs:string" minOccurs="0">
                                                         <xs:annotation>
                                                             <xs:documentation>Transaction (flight) date</xs:documentation>
                                                         </xs:annotation>
                                                     </xs:element>
-                                                    <xs:element name="flight-number" minOccurs="0">
+                                                    <xs:element name="flight-number" type="xs:string" minOccurs="0">
                                                         <xs:annotation>
                                                             <xs:documentation>Airline identifier and flight number</xs:documentation>
                                                         </xs:annotation>
                                                     </xs:element>
-                                                    <xs:element name="departure-city" minOccurs="0">
+                                                    <xs:element name="departure-city" type="xs:string" minOccurs="0">
                                                         <xs:annotation>
                                                             <xs:documentation>Departure city code</xs:documentation>
                                                         </xs:annotation>
                                                     </xs:element>
-                                                    <xs:element name="arrival-city" minOccurs="0">
+                                                    <xs:element name="arrival-city" type="xs:string" minOccurs="0">
                                                         <xs:annotation>
                                                             <xs:documentation>Arrival city code</xs:documentation>
                                                         </xs:annotation>
@@ -387,7 +387,7 @@
                                                             <xs:documentation>Card brand name</xs:documentation>
                                                         </xs:annotation>
                                                     </xs:element>
-                                                    <xs:element name="card-pan" minOccurs="0">
+                                                    <xs:element name="card-pan" type="xs:string" minOccurs="0">
                                                         <xs:annotation>
                                                             <xs:documentation>Secured card primary account number (masked or truncated)</xs:documentation>
                                                         </xs:annotation>
diff --git a/SelfServiceCommon/Massai/idl/Massai/platform.idl b/SelfServiceCommon/Massai/idl/Massai/platform.idl
index d51fb340..57b5d025 100644
--- a/SelfServiceCommon/Massai/idl/Massai/platform.idl
+++ b/SelfServiceCommon/Massai/idl/Massai/platform.idl
@@ -27,7 +27,7 @@
 
 module PlatformManagement
 {
-    const string interface_version = "2.0";
+    const string interface_version = "3.0";
 
     /** Data type for text lists. */
     typedef sequence<string> list;
@@ -165,6 +165,14 @@ module PlatformManagement
     {
       DataStatusExtension dataStatusSupport;     /* cuss datastatus supported by application */
       PaymentCardExtension paymentCardExtension; /* default media type for card reader components for the application */
+      string airlineCode;                        /* 3-letter ICAO airline code */
+      string applicationName;                
+    };
+    
+    struct PlatformEnvironment
+    {
+      string dataPath;
+      string programPath; 
     };
 
 
@@ -330,8 +338,10 @@ module PlatformManagement
         void dataReceived(in string componentType, in any data);
         
         void setupVCompListener(in string appRef, in string compRef, in any action, in any eventAcquireFilter, in any eventListener, in any elud);
-		
-		void logCounterLevel(in string unitID, in LoggingClass loggingClass, in long currentNoOfDocs, in long mediaStatus);
+
+        void logCounterLevel(in string unitID, in LoggingClass loggingClass, in long currentNoOfDocs, in long mediaStatus);
+        
+        PlatformEnvironment getEnvironment();
     };
 };
 
-- 
2.41.0.windows.1

